overlay/lib/fetch.nix

```diff
@@ -35,7 +35,7 @@ rec {
   fetchCrateAlternativeRegistryExpensive = { index, name, version, sha256 }: with buildPackages; stdenvNoCC.mkDerivation {
     name = "${name}-${version}.tar.gz";
 
-    inherit (fetchgitPrivate { url = git://dummy; }) GIT_SSH SSH_AUTH_SOCK;
+    inherit (builtins.fetchGit { url = git://dummy; }) GIT_SSH SSH_AUTH_SOCK;
```

Comment:
This expression won't actually evaluate since it's done at evaluation time, unfortunately. We need to either inherit `GIT_SSH` and `SSH_AUTH_SOCK` explicitly using `builtins.getEnv` or rewrite `fetchCrateAlternativeRegistryExpensive` to fetch the registry index using `builtins.fetchGit` properly. CC @trha

Reply:
This expression won't evaluate successfully since it's done at evaluation time, and `builtins.fetchGit` doesn't have `GIT_SSH` nor `SSH_AUTH_SOCK` attributes, AFAIK.

We need to either inherit `GIT_SSH` and `SSH_AUTH_SOCK` explicitly using `builtins.getEnv` or rewrite `fetchCrateAlternativeRegistryExpensive` to fetch the registry index using `builtins.fetchGit` properly. CC @trha